### PR TITLE
[Reader] `read/streams/discover` endpoint integration

### LINKED ISSUE #19572: [Reader IA] Jetpack integration with read/streams/discover endpoint
Context: paYKcK-3qS-p2#comment-2469

### PR DESCRIPTION
**Pull request by RenanLukas**
Fixes #19572 

-----

## To Test:

1. [App] Install JP and sign in.
2. [App] Make sure `reader_discover_new_endpoint` is enabled (Open "Me" screen -> Debug settings -> Remote Features).
3. [Android Studio] Open `App Inspection` in Android Studio.
4. [Android Studio] Open "Network Inspector".
5. [App] Open "Reader" and select "Discover".
6. 🔍  [Android Studio] Verify `read/streams/discover` is shown in "Network Inspector". Pull-to-refresh should also make a call to this endpoint. The cards should be shown as expected.
7. [App] Repeat step 2 but this time disable `reader_discover_new_endpoint`.
8. [App] Open "Reader" and select "Discover".
9. 🔍  [Android Studio] Verify `read/tags/cards` is shown in "Network Inspector". Pull-to-refresh should also make a call to this endpoint. The cards should be shown as expected.

> [!IMPORTANT]  
> An extra small fix implemented in this PR is that we also verify if the first card in "Discover" feed is a recommended blogs card. If it is, we move it to be the third card (index 2). Previously we were only doing this for recommended tags card.

-----

## Regression Notes

1. Potential unintended areas of impact

    - "Discover" feed

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

3. What automated tests I added (or what prevented me from doing so)

    --

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>ReaderDiscoverNewEndpointFeatureConfig</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by RenanLukas** [comment]
The number of changes is because I had to add a mock response JSON for the new endpoint.

image

File: WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt [hunk 1]
```diff
@@ -200,10 +207,19 @@ class ReaderDiscoverLogic @Inject constructor(
...
-        var firstYouMayLikeCard: JSONObject? = null
+        var firstRecommendationCard: JSONObject? = null
         for (i in 0 until cardsJsonArray.length()) {
             val cardJson = cardsJsonArray.getJSONObject(i)
-            when (cardJson.getString(JSON_CARD_TYPE)) {
+            // We should not have a recommended blogs or interests/tags card as the first element on Discover feed.
+            val cardType = cardJson.optString(JSON_CARD_TYPE, "")
+            val isCardTypeRecommendation =
+                cardType == JSON_CARD_RECOMMENDED_BLOGS || cardType == JSON_CARD_INTERESTS_YOU_MAY_LIKE
+            val isFirstPage = discoverTasks == REQUEST_FIRST_PAGE
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt]: Is it possible to move this check outside of the `for` loop?

**Review by daniloercoli — APPROVED** [review]
The changes look good to me, and testing steps worked as expected. I've left a comment that would be good to address before merging the PR.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt (hunk 1 above)
Comment (RenanLukas) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt]: Good catch! Done.

**Pull request merged**